### ISSUE TITLE
[float8] `float8_transpose` uses `==` instead of `=` when remapping `_axiswise_dim`

### ISSUE THREAD

**Issue by GotFusion** [pytorch/ao#4444]
## Bug

[`torchao/float8/float8_ops.py:174-179`](https://github.com/pytorch/ao/blob/main/torchao/float8/float8_ops.py#L174-L179) uses `==` (comparison, result discarded) where `=` (assignment) was clearly intended:

```python
new_axiswise_dim = old_axiswise_dim
if old_axiswise_dim is not None:
    if old_axiswise_dim == 0:
        new_axiswise_dim == -1   # ← comparison, discarded
    else:
        new_axiswise_dim == 0    # ← comparison, discarded
```

`new_axiswise_dim` is never reassigned, so after `Float8TrainingTensor.transpose(0, 1)` or `.t()` the `_axiswise_dim` attribute keeps the original value (e.g. stays `0` after a transpose that should flip it to `-1`). This silently produces wrong axis metadata that downstream matmul / scale-flattening code relies on. The companion `float8_view` (`float8_ops.py:213-239`) already does the correct remap, so this one function is just the outlier.

## Repro

```python
import torch
from torchao.float8 import e4m3_dtype
from torchao.float8.float8_scaling_utils import hp_tensor_and_scale_to_float8
from torchao.float8.float8_utils import tensor_to_scale

a = torch.rand((16, 16), dtype=torch.bfloat16)
scale = tensor_to_scale(a, e4m3_dtype)
fp8 = hp_tensor_and_scale_to_float8(a, scale, e4m3_dtype, axiswise_dim=0)
assert fp8.t()._axiswise_dim == -1, (
    f"expected -1 after transpose, got {fp8.t()._axiswise_dim}"
)
# fails on current main with: expected -1 after transpose, got 0
```

Verified the bug still exists on `main` at `b4805e2e8`.

## Fix

PR #4038 — replace `==` with `=` (2 chars), plus extends `test_transpose` to assert `_axiswise_dim` on the output (the existing test only checked `_data` / `_scale`).

## Related

Sibling typo bug in the same module: #4445 (`cc1` vs `cc2` in `Float8LinearConfig.__post_init__`). Originally bundled with this one in #4443, then split for one-to-one PR auto-close.
